### PR TITLE
[SPARK-24275][SQL] Revise doc comments in InputPartition

### PR DESCRIPTION
**Pull request by gengliangwang**
## What changes were proposed in this pull request?

In #21145,  DataReaderFactory is renamed to InputPartition. 

This PR is to revise wording in the comments to make it more clear.

## How was this patch tested?

None

### CODE REVIEW

**Comment by gengliangwang** [comment]
@rdblue @cloud-fan

File: sql/core/src/main/java/org/apache/spark/sql/sources/v2/reader/InputPartition.java [hunk 1]
```diff
@@ -42,9 +43,9 @@
    *
    * Note that if a host name cannot be recognized by Spark, it will be ignored as it was not in
    * the returned locations. By default this method returns empty string array, which means this
-   * task has no location preference.
+   * data reader has no location preference.
```
Comment (cloud-fan) [inline sql/core/src/main/java/org/apache/spark/sql/sources/v2/reader/InputPartition.java]: input partition

Comment (cloud-fan) [reply sql/core/src/main/java/org/apache/spark/sql/sources/v2/reader/InputPartition.java]: or `this input partition's reader`

**Comment by cloud-fan** [comment]
lgtm except one comment

**Comment by cloud-fan** [comment]
thanks, merging  to master!

**Pull request closed without merging**